src/index.js

```diff
@@ -0,0 +1,165 @@
+const state = {
+  location: '',
+  lat: 0,
+  lon: 0,
+  temp: 32,
+  sky: '',
+};
+
+const getLonLat = () => {
+  axios
+    .get('https://weather-report-proxy-server.herokuapp.com/location', {
+      params: {
+        q: state.location,
+        format: 'json',
+      },
+    })
+    .then((response) => {
+      console.log(response);
+      state.lon = response.data[0].lon;
+      state.lat = response.data[0].lat;
+      getLocationWeather();
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+const getLocationWeather = () => {
+  axios
+    .get('https://weather-report-proxy-server.herokuapp.com/weather', {
+      params: {
+        lat: state.lat,
+        lon: state.lon,
+      },
+    })
+    .then((response) => {
+      condition = response.data.current.weather[0].main;
+      validateSkyCondition(condition);
+      const kelvin = response.data.current.temp;
+      const fTemp = Math.floor(((kelvin - 273.15) * 9) / 5 + 32);
+      state.temp = fTemp;
+      document.getElementById('rangeValue').textContent = fTemp;
+      document.getElementById('range').value = fTemp;
+      changeColors();
+    })
+    .catch((error) => {
+      console.log('error');
+    });
+};
+
+function clickBtn(event) {
+  const city = document.getElementById('city').value;
+  state.location = city;
+  getLonLat();
+}
+
+const inputBox = document.getElementById('city');
+inputBox.onkeyup = function () {
+  document.getElementById('city-name').innerHTML = inputBox.value;
+};
+
+function rangeSlide(event) {
+  const value = event.currentTarget.value;
+  document.getElementById('rangeValue').textContent = value;
+}
+
+const validateSkyCondition = (condition) => {
+  console.log(condition);
+  if (condition === 'Clear') {
+    state.sky.value = 'sunny';
+    document.getElementById('skySelect').value = '#7496C9';
+  } else if (
+    condition === 'Rain' ||
+    condition === 'Drizzle' ||
+    condition === 'Thunderstorm'
+  ) {
+    state.sky.value = 'rainy';
+    document.getElementById('skySelect').value = '#356098';
+  } else if (condition === 'Clouds' || condition === 'Haze') {
+    state.sky.value = 'cloudy';
+    document.getElementById('skySelect').value = '#557BB1';
+  } else if (condition === 'Snow') {
+    state.sky.value = 'snowy';
+    document.getElementById('skySelect').value = '#234772';
+  }
+  skyChange();
+};
```

Comment:
I LOVED this enhancement, Hailey! It's so cool to see you develop an app based on your very own designs. I remember you being nervous about this feature during a 1:1 and I'm so proud of you for making it work ✨ YOU did that ❤️ 🔥 !!!